workbench-core/example/infrastructure/src/example-hosting-stack.ts

```diff
@@ -0,0 +1,149 @@
+
+export interface ExampleHostingStackProps extends StackProps {
+  mainAccountId: string;
+  lambdaRoleName: string;
+}
+
+export class ExampleHostingStack extends Stack {
+  public constructor(scope: Construct, id: string, props: ExampleHostingStackProps) {
+    super(scope, id, props);
+
+    const exampleHostingAccessLogsPrefix: string = 'example-hosting-access-log';
+
+    const accessLogsBucket: Bucket = createAccessLogsBucket(
+      this,
+      'ExampleHostingS3AccessLogsBucket',
+      exampleHostingAccessLogsPrefix,
+      'ExampleHostingAccessLogsNameOutput'
+    );
+
+    const exampleHostingEncryptionKey: WorkbenchEncryptionKeyWithRotation =
+      new WorkbenchEncryptionKeyWithRotation(this, 'ExampleHostingDataSetBucketEncryptionKey', {
+        removalPolicy: RemovalPolicy.DESTROY
+      });
+    const encryptionKey: Key = exampleHostingEncryptionKey.key;
+
+    const examplehostingDataSetBucket: Bucket = new WorkbenchSecureS3Bucket(
+      this,
+      'ExampleHostingDataSetBucket',
+      {
+        encryptionKey: encryptionKey,
+        serverAccessLogsBucket: accessLogsBucket,
+        serverAccessLogsPrefix: exampleHostingAccessLogsPrefix,
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true
+      }
+    ).bucket;
+
+    const exampleCrossAccountRole = new Role(this, 'ExampleCrossAccountRole', {
+      roleName: 'ExampleCrossAccountRole',
+      assumedBy: new ArnPrincipal(
+        `arn:${Aws.PARTITION}:iam::${props.mainAccountId}:role/${props.lambdaRoleName}`
+      )
+    });
+
+    exampleCrossAccountRole.addToPolicy(
+      new PolicyStatement({
+        actions: [
+          's3:GetObject',
+          's3:GetObjectVersion',
+          's3:GetObjectTagging',
+          's3:AbortMultipartUpload',
+          's3:ListMultipartUploadParts',
+          's3:GetBucketPolicy',
+          's3:PutBucketPolicy',
+          's3:PutObject',
+          's3:PutObjectAcl',
+          's3:PutObjectTagging',
+          's3:ListBucket',
+          's3:PutAccessPointPolicy',
+          's3:GetAccessPointPolicy',
+          's3:CreateAccessPoint',
+          's3:DeleteAccessPoint'
+        ],
+        resources: [
+          examplehostingDataSetBucket.bucketArn,
+          examplehostingDataSetBucket.arnForObjects('*'),
+          `arn:${Aws.PARTITION}:s3:${this.region}:${this.account}:accesspoint/*`
+        ]
+      })
+    );
+
+    //CFN NAG Suppression
+    const exampleCrossAccountRoleNode = this.node.findChild('ExampleCrossAccountRole');
+    const exampleCrossAccountRoleMetaDataNode = exampleCrossAccountRoleNode.node.findChild(
+      'Resource'
+    ) as CfnResource;
+    exampleCrossAccountRoleMetaDataNode.addMetadata('cfn_nag', {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      rules_to_suppress: [
+        // Resource found with an explicit name, this disallows updates that require replacement of this resource
```

Comment:
Not sure what updates this disallows. How does this affect the ability to update the role in the future? I suspect we will make a handful of changes to this as we move forward, then it will be stable for a while however, I want to limit how painful that handful of changes is, and I don't want changes to be complex enough so we forget how to correctly deploy them in the future.

Reply:
@DaveMadsen I need the Role name to be hardcoded in order to follow the least privilege principle. The other option is to make the permissions more open. Considering this is an integration test environment, we can always delete and redeploy the Role if deployments get affected.

Also are we expecting any more changes to the permission policy, this role is specifically to allow Cross account S3 Data Access

Reply:
For any role, I see a possible initial flurry of changes, a possible appsec related change later, then occasional (at most quarterly) changes for feature updates. I seems like this may be overkill here so I will just resolve this conversation.